test/cbmc/proofs/MQTTAgent_Terminate/MQTTAgent_Terminate_harness.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * coreMQTT-Agent v1.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* MQTT agent include. */
+#include "mqtt_agent.h"
+#include "mqtt_agent_cbmc_state.h"
+
+void harness()
+{
+    MQTTAgentContext_t * pMqttAgentContext;
+    CommandInfo_t * pCommandInfo;
+
+    pMqttAgentContext = allocateMqttAgentContext( NULL );
+    __CPROVER_assume( isValidMqttAgentContext( pMqttAgentContext ) );
+
+    pCommandInfo = malloc( sizeof( CommandInfo_t ) );
+
+    MQTTAgent_Terminate( pMqttAgentContext,
```

Comment:
Is there anything we can assert after the function returns?

Reply:
I don't think an assert on the function return adds value in this proof as the memory safety will be proven for the implementation of the function with this harness. Please let me know if you think otherwise.

Reply:
As discussed offline, I have added the asserts for the return.